### PR TITLE
Add Tools role for installing tools. See the readme for usage details…

### PR DESCRIPTION
**Pull request by afgane**
…. What's still missing is the exact toolset for the appropriate server (see files/galaxy/test.galaxyproject.org/tool_list.yaml and files/galaxy/usegalaxy.org/tool_list.yaml) and the admin user's API key for each of the servers in the vault.

### CODE REVIEW

File: files/galaxy/usegalaxy.org/tool_list.yaml [hunk 1]
```diff
@@ -0,0 +1,1058 @@
...
+  owner: 'devteam'
+  tool_panel_section_id: 'gatk'
+  revisions:
+  - '738bc749f9a3'
+- name: 'table_recalibration'
+  owner: 'devteam'
+  tool_panel_section_id: 'gatk'
+  revisions:
+  - '30e1dd77e99c'
+# Cloud-only tools
```
Comment (natefoo) [inline files/galaxy/usegalaxy.org/tool_list.yaml]: Are these supposed to be in Main's `tool_list.yaml`?

**Comment by afgane** [comment]
Looked like it to me:
https://github.com/galaxyproject/usegalaxy-playbook/blob/master/files/galaxy/usegalaxy.org/var/integrated_tool_panel.xml#L646
https://github.com/galaxyproject/usegalaxy-playbook/blob/master/files/galaxy/usegalaxy.org/var/integrated_tool_panel.xml#L647
https://github.com/galaxyproject/usegalaxy-playbook/blob/master/files/galaxy/usegalaxy.org/var/integrated_tool_panel.xml#L648

That file definitely needs more eyes and probably some tweaking. Once Test is updated, I'll start working on a script to generate a file like that directly from (any) Galaxy with it's current toolset. Once that's done, it should be easier to verify what's actually available on a Galaxy instance and in the tools file at a given point in time.

**Comment by bgruening** [comment]
@afgane maybe this is of any help for you: https://github.com/bgruening/notebooks/blob/master/galaxy/create_usegalaxy.org_dockerfile.ipynb

It tries to create a Dockerfile from all installed tools of an arbitrary Galaxy server.

File: files/galaxy/usegalaxy.org/tool_list.yaml [hunk 2]
```diff
@@ -0,0 +1,1058 @@
...
+  tool_shed_url: https://testtoolshed.g2.bx.psu.edu
+- name: data_manager_bowtie2_index_builder
+  owner:  devteam
+  tool_shed_url: https://toolshed.g2.bx.psu.edu
+- name: data_manager_bwa_mem_index_builder
+  owner: devteam
+  tool_shed_url: https://toolshed.g2.bx.psu.edu
+- name: data_manager_picard_index_builder
+  owner: devteam
+  tool_shed_url: https://testtoolshed.g2.bx.psu.edu
```
Comment (bgruening) [inline files/galaxy/usegalaxy.org/tool_list.yaml]: This is also available from the MTS.
Imho we should have a look at all TTS tools and migrate them/fix them.

Comment (martenson) [inline files/galaxy/usegalaxy.org/tool_list.yaml]: good catch, I think we should not be installing anything from TTS on machines/images we want to actually use (as opposed to testing)

Comment (bgruening) [inline files/galaxy/usegalaxy.org/tool_list.yaml]: :+1:

**Comment by afgane** [comment]
Grabbing a list of installed tools from Galaxy instance should be operational now so it's possible to to get an exact list of all revisions that are installed via the `install_tool_shed_tools.py`: 
```
gi = galaxy_instance('https://test.galaxyproject.org/', 'api key')
itl = installed_tools(gi)
print itl['shed_tools']
```
There is still a problem in that not all repositories retrieved via the tools API are also visible in the tool panel and hence it's not possible to get a tool section id for each of them. If a tool is not visible in the tools panel, is it basically a dependency that will get installed as a byproduct of installing a visible tool (i.e., can it be omitted from the tool list file)?

**Comment by bgruening** [comment]
@afgane I don't think we can omit these hidden tools. Let's imagine I would like to transfer my old workflow which used @nekrut vcftools to a new usegalaxy-like instance. These tools needs to be available. Even if the vcftools from Anton are not visible any more, because they are hidden and replaced with IUC/devteam tools.
Can we deactivate those tools after installation via the API?

**Comment by afgane** [comment]
The problem of missing section ID's as a showstopper from actually being able to install a tool remains then. For example, here's a list of tools from Test that can't be automatically matched up with a section ID: https://gist.github.com/afgane/7ccd567bf1d1ca2b88a4
About two thirds of these are packages - will those get installed automatically if all the named tools got installed? That would mean that the remaining ~50 tools would need to be manually curated to provide the missing section ID.

**Comment by bgruening** [comment]
I would not care about `package` for the moment, these should always come with tools.
I'm not sure about the rest of the list ... cummerbund is included but is installed in `NGS: RNA Analysis` - I assume this is a bug somewhere?, tophat is probably deactivated and we can try to reproduce this via 1) installing without section ID and 2) deactivating the tool.

**Comment by afgane** [comment]
I removed the `package` tools and then manually added a number more section IDs as found in the [integrated_tool_panel](https://github.com/galaxyproject/usegalaxy-playbook/blob/master/files/galaxy/test.galaxyproject.org/var/integrated_tool_panel.xml) and we not have a fairly complete list of tools on Test [here](https://github.com/galaxyproject/usegalaxy-playbook/blob/44fdabb03e6319e0f4b913e9ee93846c6a8dddde/files/galaxy/test.galaxyproject.org/tool_list.yaml).

There are still a number of outstanding tools whose section ID I could not deduce: https://gist.github.com/afgane/56114f5a0a5c34a98fb7 Some of these are data managers and datatypes but not sure what how are suites managed?

@martenson @natefoo - can you take a look and see what should be the next step in wrapping up this tools list file?

**Pull request merged**